be/src/column/map_column.cpp

```diff
@@ -234,14 +235,29 @@ void MapColumn::remove_first_n_values(size_t count) {
 }
 
 uint32_t MapColumn::serialize(size_t idx, uint8_t* pos) {
+    DCHECK(!_keys->is_map());
     uint32_t offset = _offsets->get_data()[idx];
     uint32_t map_size = _offsets->get_data()[idx + 1] - offset;
 
     strings::memcpy_inlined(pos, &map_size, sizeof(map_size));
     size_t ser_size = sizeof(map_size);
+
+    // unstable sort keys, map keys must be unique
+    SmallPermutation perm(map_size);
+    {
+        for (uint32_t i = 0; i < map_size; i++) {
+            perm[i].index_in_chunk = offset + i;
+        }
+        Tie tie(map_size, 1);
+        std::pair<int, int> range{0, map_size};
+        auto st = sort_and_tie_column(false, _keys, SortDesc(true, true), perm, tie, range, false);
+        DCHECK(st.ok());
+    }
```

Comment:
why we should sort  map keys when serializing MapColumn, sorting is time-consuming, it should be delayed to the timepoint when it really required to sort.

Reply:
Serilized method only called when compute the aggregate's group by key or hash join, map keys need a stable order to match others, like 
```
select x from (
select map{1:1, 2:2} x 
union all 
select map{2:2, 1:1}
)
group by x
```

we need promise the group by map is a stable  data, so we need sort map keys